src/Workspaces/SharedUtilitiesAndExtensions/Workspace/Core/Extensions/SyntaxGeneratorExtensions_CreateEqualsMethod.cs

```diff
@@ -31,28 +28,47 @@ public static IMethodSymbol CreateEqualsMethod(
         ParseOptions parseOptions,
         INamedTypeSymbol containingType,
         ImmutableArray<ISymbol> symbols,
-        string localNameOpt,
+        string? localNameOpt,
         SyntaxAnnotation statementAnnotation)
     {
+        var isRecord = containingType.IsRecord;
+        var localName = localNameOpt ?? (isRecord ? GetLocalName(containingType) : null);
+        var parameterName = isRecord ? localName : ObjName;
         var statements = CreateEqualsMethodStatements(
-            factory, generatorInternal, compilation, parseOptions, containingType, symbols, localNameOpt);
-        statements = statements.SelectAsArray(s => s.WithAdditionalAnnotations(statementAnnotation));
-
-        return CreateEqualsMethod(compilation, statements);
+            factory, generatorInternal, compilation, parseOptions, containingType, symbols, localName);
+
+        return CreateEqualsMethod(
+            compilation,
+            modifiers: isRecord ? containingType.IsSealed ? DeclarationModifiers.None : DeclarationModifiers.Virtual : null,
```

Comment:
Is it easier just to have the 'null' case be explicitly passing Overrides rather than defaulting it in the method? Looks like there's only two use sites...